jdaviz/configs/specviz/helper.py

```diff
@@ -76,12 +76,15 @@ def get_spectra(self, data_label=None, subset_to_apply=None, apply_slider_redshi
         # Just to save line length
         get_data_method = self.app._jdaviz_helper.get_data
         viewer = self.app.get_viewer(self._default_spectrum_viewer_reference_name)
-        statistic = getattr(viewer, "function", None)
+        statistic = None
+        if self.app.config == "cubeviz":
+            statistic = getattr(viewer.state, "function")
```

Comment:
I wonder if the original implementation would have worked with just `viewer.state` instead of `viewer` (to be a bit more general in case a profile-viewer with collapse-function is ever added to a different/custom config?

Reply:
It turns out the spectrum viewers still have a not-`None` default `function` in the other configs, I tried.